src/plugins/intel_cpu/src/nodes/multinomial.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ie_common.h>
+#include <node.h>
+
+#include <string>
+
+#include "common/permute_kernel.h"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+class Multinomial : public Node {
+public:
+    Multinomial(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context);
+
+    void getSupportedDescriptors() override;            // done
+    void initSupportedPrimitiveDescriptors() override;  // done
+
+    bool created() const override;  // done
+
+    static bool isSupportedOperation(const std::shared_ptr<const ov::Node>& op,
+                                     std::string& errorMessage) noexcept;  // done
+
+    void prepareParams() override;  // done
+
+    bool isExecutable() const override;                   // done
+    void execute(dnnl::stream strm) override;             // done
+    void executeDynamicImpl(dnnl::stream strm) override;  // done
+    bool canBeInPlace() const override {
+        return false;
+    }  // done
+
+private:
+    static constexpr size_t PROBS_PORT = 0lu;
+    static constexpr size_t NUM_SAMPLES_PORT = 1lu;
+    static constexpr size_t OUTPUT_PORT = 0lu;
+
+    ov::element::Type_t m_convert_type = ov::element::i32;
+    bool m_with_replacement = false;
+    bool m_log_probs = false;
+    uint64_t m_global_seed = 0;
+    uint64_t m_op_seed = 0;
+
+    std::string m_errorPrefix;
+    bool m_probs_1d = false;
+    size_t m_input_elements_count = 0;
+    size_t m_batches_count = 0;
+    size_t m_probs_count = 0;
+    std::vector<float> m_input_vals;
+    std::vector<float> m_cdf;
+    std::vector<float> m_max_per_batch;
+    std::vector<double> m_random_samples;
+    std::vector<int> m_output_vals;
+
+    // std::unique_ptr<RandomUniform> random_uniform_kernel; // TODO waiting for kernel
```

Comment:
As far as I understand it, should be another random distribution here.

Reply:
RandomUniform with Phillox appears to be the one used by other frameworks, which in turn makes so that the implementations give the same result with the same seed, hence the reason why this was introduced